src/codeu/chat/server/Model.java

```diff
@@ -67,15 +68,57 @@ public int compare(Time a, Time b) {
   private final Store<Time, Message> messageByTime = new Store<>(TIME_COMPARE);
   private final Store<String, Message> messageByText = new Store<>(STRING_COMPARE);
 
+  private final HashMap<Uuid, HashMap<Uuid, Integer>> userConversationTracking = new HashMap<Uuid, HashMap<Uuid, Integer>>();
```

Comment:
This is storing state about UserFollowings outside of the User object which I am not a fan of.

Instead, I would remove this altogether.  You already have UsersByID that let's you find an User object from the UUID.  Once you have an user object, similar to UserFollowings, each User can have a collection of ConversationFollowing objects.  When the ConversationFollowing is created, it can remember the size of that conversation to use for message count calculations.